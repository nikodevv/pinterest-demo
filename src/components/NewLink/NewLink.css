.modalContainer {
  width: 70%;
  height: 40vh;
  background-color: white;
}

.modalContainer input[type="text"]{
  outline: 0;
  width: 40%;
  border-radius: 0.25rem;
  border-color: transparent;
  box-shadow: 0 0 0 0.1rem #9d9d9d;
  margin-top: 1rem;
}

.modalContainer input:focus {
  box-shadow: 0 0 0 0.25rem #91b9ff;
}

.modalContainer button {
  outline: 0;
  height: 1.5rem;
  margin-top: 1rem;
  color: #9d9d9d;
  border-radius: 0.25rem;
  border-color: transparent;
  background-color: white;
  box-shadow: 0 0 0 0.1rem #9d9d9d;
}

.modalContainer button:disabled {
  color: white;
  background-color: #9d9d9d;
}
.modalContainer button:active {
  box-shadow: 0 0 0 0.25rem #91b9ff;
}

