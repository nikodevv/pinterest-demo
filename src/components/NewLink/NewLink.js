import React from 'react';
import './NewLink.css';
import {firebaseAuth} from "../../utility/firebaseFascade";
import {modalActionCreators} from "../../actions";
import {useDispatch} from "react-redux";

const helpers = {
  saveLink: (imgUrl, linkUrl, dispatch)=>{
    firebaseAuth.addPost({
      imgUrl,
      linkUrl
    });
    dispatch(modalActionCreators.toggleNewLinkModal())
  }
};

const NewLink = () => {
  const [imgUrl, setImgUrl] = React.useState('');
  const [linkUrl, setLinkUrl] = React.useState('');
  const dispatch = useDispatch();
  return <div className="screenOverlay container-row center"
              onClick={(e) => {
                if (e.currentTarget === e.target){
                  dispatch(modalActionCreators.toggleNewLinkModal());
                }
              }}>
    <div className="container-column center modalContainer" onClick={(e)=>e.preventDefault()}>
      <div className="center container-row">
        <h1>Enter a new post</h1>
      </div>
      <div className="center container-row">
        <input placeholder="image url" type="text" onChange={(e)=>setImgUrl(e.target.value)}/>
      </div>
      <div className="center container-row">
        <input placeholder="link location" type="text" onChange={(e)=>setLinkUrl(e.target.value)}/>
      </div>
      <div className="center container-row">
        <button disabled={(!linkUrl || !imgUrl)}
                onClick={(e)=>{{
                  e.preventDefault();
                  helpers.saveLink(imgUrl, linkUrl, dispatch)
                }}}>
          Save
        </button>
      </div>
    </div>
  </div>;
};

export default NewLink;
