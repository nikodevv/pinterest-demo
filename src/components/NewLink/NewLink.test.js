import React from 'react';
import {configure, shallow} from 'enzyme';
import {Provider} from "react-redux";
import Adapter from "enzyme-adapter-react-16/build";
import configureMockStore from 'redux-mock-store';
import NewLink from './NewLink';

configure({adapter: new Adapter()});

const mockStore = configureMockStore();

describe('<NewLink />', () => {
  test('renders', () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <NewLink />
      </Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
