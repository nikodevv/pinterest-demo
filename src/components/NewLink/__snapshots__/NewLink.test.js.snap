// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewLink /> renders 1`] = `ShallowWrapper {}`;
